Add `TextServer.get_short_name()` and use it in `--text-driver`

Calinou opened the pull request:
- This closes https://github.com/godotengine/godot-proposals/issues/13725.
- See https://github.com/godotengine/godot/pull/111014#pullrequestreview-3281345157.

bruvzg reviewed (commented):
Instead of hard-coding it, it's probably better to add a new method to TextServer (`get_short_name()` or something like this) and use it for command line argument handling.

Calinou committed:
Add `TextServer.get_short_name()` and use it in `--text-driver`

This makes it easier to select TextServers on the command line.
Instead of having to enter the full name in quotes, you can now
just use `advanced`, `fallback` or `dummy` (lowercase).

AThousandShips reviewed (approved):
Style and docs look good!

Repiteo commented:
Thanks!
